### PR TITLE
feat(ruby): e2e CTS

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Following the other languages, add e2e tests for ruby

### CODE REVIEW

File: templates/ruby/tests/requests/requests.mustache [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 
 class Test{{#lambda.pascalcase}}{{{client}}}{{/lambda.pascalcase}} < Test::Unit::TestCase
   include Algolia::{{{modelModule}}}
   def setup
     @client = Algolia::{{#lambda.pascalcase}}{{{client}}}{{/lambda.pascalcase}}.create_with_config(
       Algolia::Configuration.new(
-      'APP_ID', 
-      'API_KEY', 
+      ENV.fetch('ALGOLIA_APPLICATION_ID', nil),
+      ENV.fetch('ALGOLIA_ADMIN_KEY', nil),
```
Comment (shortcuts) [inline templates/ruby/tests/requests/requests.mustache:13]: maybe those can stay hardcoded?

File: templates/ruby/tests/requests/requests.mustache [hunk 2]
```diff
@@ -32,6 +42,14 @@ class Test{{#lambda.pascalcase}}{{{client}}}{{/lambda.pascalcase}} < Test::Unit:
     assert(req.body.nil?, 'body is not nil')
     {{/assertNullBody}}{{/body}}
     {{/request}}
+
+    {{#response}}
+    res = @e2e_client.{{#lambda.snakecase}}{{method}}{{/lambda.snakecase}}_with_http_info({{#parametersWithDataType}}{{> requests/generateParams}}{{/parametersWithDataType}}{{#hasRequestOptions}}{ {{#requestOptions.headers.parameters}}:header_params => JSON.parse('{{{.}}}', :symbolize_names => true),{{/requestOptions.headers.parameters}}{{#requestOptions.queryParameters.parameters}}:query_params => JSON.parse('{{{.}}}', :symbolize_names => true){{/requestOptions.queryParameters.parameters}} }{{/hasRequestOptions}})
+
+    {{#statusCode}}
+    assert_equal(res.status, {{statusCode}})
+    {{/statusCode}}
```
Comment (shortcuts) [inline templates/ruby/tests/requests/requests.mustache:51]: you don't assert the body yet?

**Review by shortcuts — COMMENTED** [review]
💘

File: templates/ruby/tests/requests/requests.mustache (hunk 2 above)
Comment (millotp) [reply templates/ruby/tests/requests/requests.mustache:51]: not yet, I'm waiting to see if you're partial thing works 😛

File: templates/ruby/tests/requests/requests.mustache (hunk 1 above)
Comment (millotp) [reply templates/ruby/tests/requests/requests.mustache:13]: they can, it doesn't make a huge difference

**Review by shortcuts — APPROVED** [review]
LOVE IT

**Pull request merged**